assets/js/src/core/modules/asset/actions/clear-thumbnails/use-clear-thumbnails.tsx

```diff
@@ -0,0 +1,52 @@
+/**
+* Pimcore
+*
+* This source file is available under two different licenses:
+* - Pimcore Open Core License (POCL)
+* - Pimcore Commercial License (PCL)
+* Full copyright and license information is available in
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import { type ItemType } from '@Pimcore/components/dropdown/dropdown'
+import { type Asset, useAssetImageClearThumbnailMutation } from '@Pimcore/modules/asset/asset-api-slice.gen'
+import { Icon } from '@Pimcore/components/icon/icon'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+
+export interface UseClearThumbnailsHookReturn {
+  clearThumbnailContextMenuItem: (node: Asset, onFinish?: () => void) => ItemType
+}
+
+export const useClearThumbnails = (): UseClearThumbnailsHookReturn => {
+  const { t } = useTranslation()
+  const [clearThumbnail] = useAssetImageClearThumbnailMutation()
+
+  const handleClearThumbnails = async (node: Asset, onFinish?: () => void): Promise<void> => {
+    const clearThumbnailTask = clearThumbnail({ id: node.id })
+
+    try {
+      await clearThumbnailTask
+      onFinish?.()
+    } catch (error) {
+      console.error('Error clearing thumbnails (id: ' + node.id + ')', error)
+    }
+  }
+
+  const clearThumbnailContextMenuItem = (node: Asset, onFinish?: () => void): ItemType => {
+    return {
```

Comment:
For permission and hidden checks take a look here:
https://github.com/pimcore/admin-ui-classic-bundle/blob/2.x/public/js/pimcore/asset/asset.js#L316

We need to do the same checks like in the classic bundle. E.g. it should also work for videos - does this work on backend side?

Reply:
improved with https://github.com/pimcore/studio-ui-bundle/pull/869/commits/cc3b3b0f3d17005d929260add6885b92a2123b5b

I am currently waiting for the correct icon.